From e7f6f2a9af0c53a1e5e512e63268a0aac6f3a35b Mon Sep 17 00:00:00 2001
From: Minecrell <minecrell@minecrell.net>
Date: Sat, 29 Feb 2020 13:22:44 +0100
Subject: [PATCH] android-base/endian.h: fix build on musl

So far musl has used the definitions for Windows. Now that it includes
<winsock2.h> this fails horribly. Add a proper fallback for !__GLIBC__
but __linux__ with defines for musl.
---
 base/include/android-base/endian.h | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/base/include/android-base/endian.h b/base/include/android-base/endian.h
index 1e5cceb9a..c85c0ae30 100644
--- a/base/include/android-base/endian.h
+++ b/base/include/android-base/endian.h
@@ -44,6 +44,17 @@
 #define letoh32(x) le32toh(x)
 #define letoh64(x) le64toh(x)
 
+#elif defined(__linux__)
+/* musl's <endian.h> is like bionic's <sys/endian.h>. */
+#include <endian.h>
+
+/* musl keeps htons and htonl in <netinet/in.h>. */
+#include <netinet/in.h>
+
+/* musl doesn't have the 64-bit variants. */
+#define htonq(x) htobe64(x)
+#define ntohq(x) be64toh(x)
+
 #else
 
 #if defined(__APPLE__)
-- 
2.25.1

